js/app/dashboard/redux/actions.js

```diff
@@ -292,63 +304,47 @@ export function importError(token, message) {
   return { type: IMPORT_ERROR, token, message }
 }
 
-export function modifyTaskListInUI(username, tasks) {
-  /*
-    Modify a TaskList
-  */
-
-  return function(dispatch, getState) {
-
-    let state = getState()
-    let allTasks = selectAllTasks(state)
-
-    const newTasks = tasks.map((task, position) => {
-      const rt = _.find(allTasks, t => t['@id'] === task['@id'])
-
-      return {
-        ...rt,
-        isAssigned: true,
-        position,
-      }
-    })
-    const tasksLists = selectTaskLists(getState())
-    const tasksList = _.find(tasksLists, tl => tl.username === username)
-    const previousTasks = tasksList.items
-
-    return dispatch(modifyTaskListRequest(username, newTasks, previousTasks))
-  }}
-
-export function modifyTaskList(username, tasks) {
-  /*
-    Modify a TaskList
-  */
+/**
+ * Modify a TaskList
+ * @param {string} Username - Username of the rider to which we assign
+ * @param {Array.Objects} items - Items to be assigned, list of tasks and tours to be assigned
+ */
+export function modifyTaskList(username, items) {
 
   return async function(dispatch, getState) {
 
     const state = getState()
 
-    dispatch(modifyTaskListInUI(username, tasks))
+    const tasksList = selectTaskListByUsername(getState(), {username: username})
+    const previousItems = tasksList.items
 
-    const data = tasks.map((task, index) => ({
-      task: task['@id'],
-      position: index,
-    }))
+    // support passing URIs directly - TODO uniformize behaviour
+    const newItems = items.map((item) => item['@id'] || item)
+
+    dispatch(modifyTaskListRequest(username, newItems, previousItems))
 
     const date = selectSelectedDate(state)
 
-    const url = window.Routing.generate('admin_task_list_modify', {
+    const url = window.Routing.generate('api_task_lists_set_items_item', {
       date: date.format('YYYY-MM-DD'),
       username,
     })
 
+    const { jwt } = getState()
+    const httpClient = createClient(dispatch)
```

Comment:
Did you see another http client implementation in `user-data.js`? I think we should standardise on one implementation at some point

Reply:
https://github.com/coopcycle/coopcycle-web/issues/4338